### PR TITLE
Vulkan: Use v2 feature checks, and add check for multiview features.

### PR DESCRIPTION
**Pull request by hrydgard**
Extracted from the multiview PR #16273, and renamed some stuff.

Quite a few lines of change, so it's nice to have separate for review purposes.

### CODE REVIEW

File: ext/vulkan/vulkan_core.h [hunk 1]
```diff
@@ -4980,7 +4980,7 @@ typedef struct VkSparseImageMemoryRequirements2 {
 typedef struct VkPhysicalDeviceFeatures2 {
     VkStructureType             sType;
     void*                       pNext;
-    VkPhysicalDeviceFeatures    features;
+    VkPhysicalDeviceFeatures    standard;
```
Comment (unknownbrackets) [inline ext/vulkan/vulkan_core.h:4983]: I don't mind this, but should we actually change this header file?  We could still call our struct standard either way.

-[Unknown]

Comment (hrydgard) [reply ext/vulkan/vulkan_core.h:4983]: Oops, that was not intentional, heh. A global rename went haywire and I misse this damage.

**Review by unknownbrackets — APPROVED** [review]

**Comment by hrydgard** [comment]
Huh, this randomly failed shadergen test points to an interesting issue with fragment shader writes:

https://github.com/hrydgard/ppsspp/actions/runs/3308596568/jobs/5461171115

we write stuff like rounded depth at the very end, but that might be after discard. And not writing gl_FragDepth on some paths is not allowed!

**Comment by unknownbrackets** [comment]
Oh, that's interesting.  But even the depth write should be discarded right?  I suppose we should correctly write to it always, though, perhaps?  Wouldn't this be similarly true for fragColor1/fragColor0?

-[Unknown]

**Comment by hrydgard** [comment]
Yeah, you'd think so, I dunno what's going on really... 

```
  31:  PS_OUT main(  vec2 v_texcoord : TEXCOORD0) {
  32:    PS_OUT ps_out;
  33:    float gl_FragDepth;
  34:    vec4 index = tex.Sample(texSamp, v_texcoord.xy);
  35:    vec4 outColor = index.aaaa;
  36:    float shifted = roundAndScaleTo255f(index.a) / roundAndScaleTo255f(stencilValue);
  37:    if (mod(floor(shifted), 2.0) < 0.99) DISCARD;
  38:    ps_out.target = outColor;
  39:    ps_out.depth = gl_FragDepth;
  40:    return ps_out;
  41:  }
```

That is a pretty broken shader though because nothing is ever written to our temporary gl_FragDepth.

```
errors: D:\a\ppsspp\ppsspp\Shader@0x000002B132E71FD0(31,11-42): warning X3578: Output value 'main' is not completely initialized
D:\a\ppsspp\ppsspp\Shader@0x000002B132E71FD0(31,11-42): error X4580: Output variable main contains a system-interpreted value (SV_Depth) which must be written in every execution path of the shader.  Unconditional initialization may help.
```

I'll at least clean that up tomorrow.

**Comment by unknownbrackets** [comment]
"Error compiling stencil shader" - the stencil shader should only write depth when using the Adreno bug workaround, right?  I don't get what changed in this pull request to affect that, though.

-[Unknown]

**Comment by hrydgard** [comment]
Oh never mind it's just an uninitialized variable.

**Comment by unknownbrackets** [comment]
Actually seems it's caused by #16278, but somehow its checks initially passed until it merged into master.  Maybe dependent on a change already in master to trigger...

-[Unknown]

**Pull request merged**